### ISSUE TITLE
Feature request: Programmatically navigate via utility function so I can use it in redux-saga

### ISSUE THREAD

**Issue by rsml** [nandorojo/solito#215]
I'm migrating a project to solito, and I can't find a clean way to trigger a navigation to a new screen from within a redux-saga generator function. Neither `<Link>` nor `useRouter()` are suited for this (since I want my navigations to be triggered asynchronously and not within a react component).

Prior to solito I would would use this utility function inside my sagas

```javascript
export function navigate(routeName, params) {
  _navigator.current.dispatch(
    CommonActions.navigate({
      name: routeName,
      params,
    }),
  );
}
```

(Where `_navigator` was a ref to my `<NavigationContainer>` from `@react-navigation/native`)

I can do something like this as a hacky work around. But it would require me to implement `urlForRouteName` and to come up with some solution for handling params:
```javascript
export function navigate(routeName, params) {
  if(Platform.OS === 'web') {
    window.location.href = urlForRouteName[routeName]
  } else {
    _navigator.current.dispatch(
      CommonActions.navigate({
        name: routeName,
        params,
      }),
    );
  }
}
```

So in short, my suggestion is to provide utility functions around all of the CommonActions that are defined by React Navigation (https://reactnavigation.org/docs/navigation-actions/):
* `'GO_BACK'`
* `'NAVIGATE'`
* `'RESET'`
* `'SET_PARAMS'`

Thanks in advance for any help. And thank you very much for solito.

**Comment by nandorojo** [comment]
Hmm well for starters on Web I would use `import Router from ‘next/router’`. Not entirely sure about the native side if you don’t have the context. Maybe you could use `getPathFromState`. But generally the functions are meant to be used in context

**Comment by rsml** [comment]
Ok thanks again for all your work on this library! And thanks for the quick response. I decided to just move all of my navigation code away from the sagas to simplify things. Closed.
